importers/link_16_importer.py

```diff
@@ -0,0 +1,199 @@
+        # Read the base timestamp from the filename
+        filename, ext = os.path.splitext(os.path.basename(path))
+        datetime_string = self.extract_timestamp(filename)
+        self.base_timestamp = self.timestamp_to_datetime(datetime_string)
+        self.current_hour = 0
+        self.previous_time = timedelta(hours=0, minutes=0, seconds=0)
+
+        # Now do what we'd normally do on load
+        for line_number, line in enumerate(tqdm(file_object.lines()), 1):
+            result = self._load_this_line(data_store, line_number, line, datafile, change_id)
+            if result == CANCEL_IMPORT:
+                custom_print_formatted_text(
+                    format_error_message(f"Error in file caused cancellation of import of {path}")
+                )
+                break
+
+    def _load_this_line(self, data_store, line_number, line, datafile, change_id):
+        if line_number == 1:
+            # Will only parse if one of these headers found
+            self.version = 1 if line.text.startswith(V1_HEADER) else 2
+            return
+        tokens = line.tokens(line.CSV_TOKENISER, ",")
+        if len(tokens) <= 1:
+            # To skip over any blank lines
+            return
+
+        # Time wrangling - TODO - consider extracting into a method
+        time_token = tokens[1]
+        line_time = timedelta(
+            hours=0,
+            minutes=int(time_token.text[:2]),
+            seconds=float(time_token.text[3:]),
+        )
+
+        if line_time < self.previous_time:
+            self.current_hour += 1
+
+        self.previous_time = line_time
+        line_time += timedelta(hours=self.current_hour)
+        line_time += self.base_timestamp
+
+        if self.version == 1:
+            if len(tokens) < 15:
+                self.errors.append(
+                    {self.error_type: f"Error on line {line_number}. Not enough tokens: {line}"}
+                )
+                return
+            name_token = tokens[2]
+            lat_degrees_token = tokens[8]
+            lon_degrees_token = tokens[7]
+            heading_token = tokens[11]
+            speed_token = tokens[12]
+            altitude_token = tokens[10]
+
+        elif self.version == 2:
+            if len(tokens) < 28:
+                self.errors.append(
+                    {self.error_type: f"Error on line {line_number}. Not enough tokens: {line}"}
+                )
+                return
+            name_token = tokens[4]
+            lat_degrees_token = tokens[13]
+            lon_degrees_token = tokens[14]
+            heading_token = tokens[16]
+            speed_token = tokens[17]
+            altitude_token = tokens[15]
+
+        # -- TODO - Confirm that this is correct for this data
+        platform = self.get_cached_platform(
+            data_store, platform_name=name_token.text, change_id=change_id
+        )
+        # TODO - confirm the actual sensor type
```

Comment:
this may be getting philosophical, but I "think" the sensor type is GPS.  Platforms are reporting their own position via Link-16, and since they're airborne the sensor used to determine their position is GPS.

Reply:
That makes sense. Worth confirming, but you've convinced me 😃

Reply:
Discussion with MWC confirmed sensor type to be GPS